cmd/nerdctl/run_network.go

```diff
@@ -164,12 +167,87 @@ func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]
 				ports = append(ports, pm...)
 			}
 		}
+	case nettype.Container:
+		if cmd.Flags().Changed("publish") {
+			return nil, nil, "", nil, fmt.Errorf("conflicting options: port publishing and the container type network mode")
+		}
+		if runtime.GOOS != "linux" {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can only works on linux")
+		}
+		if rootlessutil.IsRootless() {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can't run rootless")
+		}
+		if len(netSlice) > 1 {
+			return nil, nil, "", nil, fmt.Errorf("only one network allowed using '--network=container:<container>'")
+		}
+		network := strings.Split(netSlice[0], ":")
+		if len(network) != 2 {
+			return nil, nil, "", nil, fmt.Errorf("invalid network: %s, should be \"container:<id|name>\"", netSlice[0])
+		}
+		containerName := network[1]
+		client, ctx, cancel, err := newClient(cmd)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		defer cancel()
+		var containerID string
+		walker := &containerwalker.ContainerWalker{
+			Client: client,
+			OnFound: func(ctx context.Context, found containerwalker.Found) error {
+				if found.MatchCount > 1 {
+					return fmt.Errorf("multiple containers found with prefix: %s", containerName)
+				}
+				containerID = found.Container.ID()
+				// save in label
+				netSlice = []string{fmt.Sprintf("container:%s", containerID)}
+				return nil
+			},
+		}
+		n, err := walker.Walk(ctx, containerName)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		if n == 0 {
+			return nil, nil, "", nil, fmt.Errorf("no such container: %s", containerName)
+		}
+
+		path, err := getNetworkNamespace(ctx, client, containerID)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+
+		opts = append(opts, oci.WithLinuxNamespace(specs.LinuxNamespace{
+			Type: specs.NetworkNamespace,
+			Path: path,
+		}))
+
 	default:
 		return nil, nil, "", nil, fmt.Errorf("unexpected network type %v", netType)
 	}
 	return opts, netSlice, ipAddress, ports, nil
 }
 
+func getNetworkNamespace(ctx context.Context, client *containerd.Client, containerID string) (string, error) {
```

Comment:
To be honest, this is not a good function name, because it returns a `can't join network` error.